codebase/src/components/shapes.ts

```diff
@@ -0,0 +1,52 @@
+import type { GEdge, GNode } from "@/graph/types";
+
+type GraphObject = {
+	data: GEdge | GNode,
+	isIntersecting: (data: GEdge | GNode, mouseX: number, mouseY: number) => boolean,
+};
+
+// Implement a shape that has two parts draw and is intersecting
+// Each object on the graph has its data, priority and shape associated with it
```

Comment:
use jsdoc for this kind of stuff. You will get much better in-editor documentation this way